### DISCUSSION TITLE
Multiple UMAPs

### DISCUSSION THREAD

**Discussion by rasing02** [satijalab/seurat#9466] [General]
Hello All,
I have ten groups in the scRNA study, and I want to know if there is a way to obtain the UMAP output for each group in a different PDF file. 

The code pasted below outputs all ten groups as one file
pdf("UMAP_group.pdf", width= 46 , height= 18)
DimPlot(seurat.obj, reduction = "umap", split.by= "group", label = FALSE, ncol = 5, raster=FALSE)
dev.off()

I tried setting combine = FALSE to obtain separate pdf's for each group, however this did not work. Please let me know if you have any suggestions.

Thank you!

**Comment by samuel-marsh** [comment]
Hi,

The issue is not with `combine` but with the `pdf()` call which will create on pdf for each call until `dev.off()`.

If you want to create multiple PDFs you can write things as for loop or using apply.  To save some time though you can use iterative plotting functions from my package scCustomize (https://samuel-marsh.github.io/scCustomize/articles/Iterative_Plotting.html).  See `Iterate_DimPlot_bySample`.

by sample is really just any meta.data column you choose so in your case above set `sample_column = “group”`

Best,
Sam

**Reply by rasing02** [reply]
Thank you Samuel

**Reply by rasing02** [reply]
Dimplot function works but Iterate is not. I used the following code, but it leads to empty files. It creates an x and y axis for UMAP.
Iterate_DimPlot_bySample(
  seuraobj,
  sample_column = "group",
  file_path = "plots/",
  file_name = "umap",
  file_type = ".jpeg",
  #single_pdf = FALSE,
  #color = "black",
  dpi = 600,
  no_legend = TRUE,
  title_prefix = NULL,
  reduction = "umap",
  #pt.size = NULL,
  #raster = NULL,
  
)

**Reply by samuel-marsh** [reply]
Can you post to scCustomize repo so I can better track and help with this?

thanks,
Sam

**Reply by rasing02** [reply]
I just posted to scCustomize repo. Thank you!
